FrameworkSystem/Service/SystemAdministratorHandler.py

```diff
@@ -54,7 +55,12 @@ def initializeHandler( cls, serviceInfo ):
       global gMonitoringReporter
       gMonitoringReporter = MonitoringReporter( monitoringType = "ComponentMonitoring" )
       gThreadScheduler.addPeriodicTask( 120, cls.__storeProfiling )
-      
+    
+    keepSoftwareVersions = cls.srv_getCSOption( 'KeepSoftwareVersions', 0 )
+    if keepSoftwareVersions:
+      gLogger.info( "The last %s software version will be kept and the rest will be deleted!" % keepSoftwareVersions )
+      gThreadScheduler.addPeriodicTask( 120, cls.__deleteOldSoftware, ( keepSoftwareVersions, ), executions = 2 ) #it is enough to try 2 times
```

Comment:
I think we can do this cleaning once a day... every 2 minutes is a bit aggressive